test/Riok.Mapperly.Tests/Generator/IncrementalGeneratorTest.cs

```diff
@@ -0,0 +1,161 @@
+{
+    [Fact]
+    public void AddingUnrelatedTypeDoesNotRegenerateOriginal()
+    {
+        var source = TestSourceBuilder.CSharp(
+            """
+            using Riok.Mapperly.Abstractions;
+
+            namespace Test.A
+            {
+                [Mapper]
+                internal partial class FooBarMapper
+                {
+                    internal partial string FooToBar(string value);
+                }
+            }
+            """
+        );
+
+        var syntaxTree = CSharpSyntaxTree.ParseText(source, CSharpParseOptions.Default);
+        var compilation1 = TestHelper.BuildCompilation(TestHelperOptions.NoDiagnostics.NullableOption, syntaxTree);
+
+        var driver1 = TestHelper.TrackingGeneratorDriver().RunGenerators(compilation1);
+
+        var compilation2 = compilation1.AddSyntaxTrees(CSharpSyntaxTree.ParseText("struct MyValue {}", CSharpParseOptions.Default));
+        var driver2 = driver1.RunGenerators(compilation2);
+
+        AssertRunResults(driver2, IncrementalStepRunReason.Cached);
+    }
+
+    [Fact]
+    public void AddingNewMapperDoesNotRegenerateOriginal()
+    {
+        var source = TestSourceBuilder.CSharp(
+            """
+            using Riok.Mapperly.Abstractions;
+
+            namespace Test.A
+            {
+                [Mapper]
+                internal partial class FooBarMapper
+                {
+                    internal partial string FooToBar(string value);
+                }
+            }
+            """
+        );
+
+        var secondary = TestSourceBuilder.CSharp(
+            """
+            using Riok.Mapperly.Abstractions;
+
+            namespace Test.B
+            {
+                [Mapper]
+                internal partial class BarFooMapper
+                {
+                    internal partial string BarToFoo(string value);
+                }
+            }
+            """
+        );
+
+        var syntaxTree = CSharpSyntaxTree.ParseText(source, CSharpParseOptions.Default);
+        var compilation1 = TestHelper.BuildCompilation(TestHelperOptions.NoDiagnostics.NullableOption, syntaxTree);
+
+        var driver1 = TestHelper.TrackingGeneratorDriver().RunGenerators(compilation1);
+
+        var compilation2 = compilation1.AddSyntaxTrees(CSharpSyntaxTree.ParseText(secondary, CSharpParseOptions.Default));
+        var driver2 = driver1.RunGenerators(compilation2);
+
+        AssertRunResults(driver2, IncrementalStepRunReason.Cached, IncrementalStepRunReason.New);
+    }
+
+    [Fact]
+    public void AppendingUnrelatedTypeDoesNotRegenerateOriginal()
+    {
+        var source = TestSourceBuilder.CSharp(
+            """
+            using Riok.Mapperly.Abstractions;
+
+            namespace Test.A
+            {
+                [Mapper]
+                internal partial class FooBarMapper
+                {
+                    internal partial string FooToBar(string value);
+                }
+            }
+            """
+        );
```

Comment:
Instead of these mapper sources couldn't `Riok.Mapperly.Tests.TestSourceBuilder.Mapping("string", "string")` be used?